pithos/pithos.py

```diff
@@ -366,6 +368,23 @@ def get_proxy(self):
 
         return None
 
+    def on_explicit_content_filter_checkbox(self, *ignore):
+        if self.pandora.connected:
+            def fn(*ignore):
+                if self.filter_state != self.prefs_dlg.explicit_content_filter_checkbutton.get_active():
+                    self.pandora.set_explicit_content_filter_to(self.prefs_dlg.explicit_content_filter_checkbutton.get_active())
+
+            def callback(*ignore):
+                if self.prefs_dlg.explicit_content_filter_checkbutton.get_active():
+                    logging.info('Getting new playlist.')
+                    self.waiting_for_playlist = False
+                    self.stop()
+                    self.current_song_index = None
+                    self.songs_model.clear()
+                    self.get_playlist(start = True)
+
+            self.worker_run(fn, '', callback)
```

Comment:
Pass the current state as an arg here instead of touching the UI from `fn()`.

Same comment about useful function names.